code/network/multi_obj.h

```diff
@@ -32,19 +35,80 @@ struct net_player;
 #define OOC_AFTERBURNER_ON			(1<<5)
 // two spots now left for more OOC flags
 
+// what we are setting the subsystem "index" to if there was no subsystem.
+#define OOC_INDEX_NULLPTR_SUBSYSEM 255
```

Comment:
If the subsystem count ever gets bumped this could be a problem. May want to look for a different solution here (use short type to allow for -1?)

Reply:
It was bumped to a ushort, but I forgot to make the change to this constant to allow for that.  I can just use the ushort max?

Reply:
If you used a short then you should be able to assign it without having to check for this special value. And SHRT_MAX is still a bunch of subsystems.

It works either way though, so whichever you think works better in context should be fine.

Reply:
I mean, we have no idea what crazy things modders will do in the future, so future-proofing as much as possible is my goal.  Some of the things that Oddgrim and others are doing make me worried about how to fully support models of the future in multi.

Reply:
What about doing this:
```
#define OOC_INDEX_NULLPTR_SUBSYSEM static_cast<ushort>(-1);
```
Casting -1 to an unsigned type automatically makes it the highest value of that type, I think.  If the type is ever changed to unsigned int, that method should still work.